src/spinner/index.ts

```diff
@@ -108,17 +108,25 @@ export async function spinner(
   {
     start,
     end,
+    onError,
     while: update = () => sleep(100),
-  }: { start: string; end: string; while: (...args: any) => Promise<any> },
+  }: { start: string; end: string; onError?: (e: any) => void; while: (...args: any) => Promise<any> },
   { stdin = process.stdin, stdout = process.stdout } = {}
 ) {
-  const act = update();
-  const tooslow = Object.create(null);
-  const result = await Promise.race([sleep(500).then(() => tooslow), act]);
-  if (result === tooslow) {
     const loading = await gradient(chalk.green(start), { stdin, stdout });
-    await act;
-    loading.stop();
-  }
-  stdout.write(`${" ".repeat(5)} ${chalk.green("✔")}  ${chalk.green(end)}\n`);
-}
+
+    try {
+        const act = update();
+        const tooslow = Object.create(null);
+        const result = await Promise.race([sleep(500).then(() => tooslow), act]);
+        if (result === tooslow) {
+            await act;
+        }
+
+        stdout.write(`${" ".repeat(5)} ${chalk.green("✔")}  ${chalk.green(end)}\n`);
+    } catch (e) {
```

Comment:
nit: I would start the `try` block after line 120, before `Promise.race`, which is where the error can occur